File: Snappet.Challenge/Models/WorkQuery.cs
using System;

namespace Snappet.Challenge.Models
{
    public class WorkQuery
    {
        public string Subject { get; set; }
        public string Domain { get; set; }
        public string LearningObjective { get; set; }
        public bool? Correct { get; set; }
        public int? User { get; set; }
        public int? Exercise { get; set; }
        public DateTime DateSubmitted { get; set; }
        public int ClientTimeZoneOffset { get; set; }
        public int PageNumber { get; set; }
        public int ItemsPerPage { get; set; }
    }
}

File: Snappet.Challenge/ClientApp/app/interfaces/work-query.ts
export interface WorkQuery {
    subject: string | null;
    domain: string | null;
    learningObjective: string | null;
    correct: boolean | null;
    user: number | null;
    exercise: number | null;
    dateSubmitted: Date | null;
    clientTimeZoneOffset: number;
    pageNumber: number;
    itemsPerPage: number;
}